### PR TITLE
Fix: “pointer to empty string” is not omitted bug of AWS X-Ray Exporter

### PR DESCRIPTION
**Pull request by EdZou**
**Description:** 
current exporter is broken because of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/691 - unfortunately `omitempty` doesn’t work because “pointer to empty string” is not omitted.

Now added a util function under "internal/common/awsxray" repository and take use of this to check the "pointer to empty string"

If you would like to check @anuraaga

### CODE REVIEW

**Comment by pjanotti** [comment]
Hi @EdZou - the change itself looks fine but could you please add a test to cover this case?

/cc @kbrockhoff @anuraaga

**Comment by EdZou** [comment]
Hello @pjanotti I just added unit test, if you would like to review
cc: @kbrockhoff @anuraaga

File: internal/common/awsxray/util.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsxray
+
+// Util to check nil string pointer
```
Comment (anuraaga) [inline internal/common/awsxray/util.go:17]:
```suggestion
// String returns a pointer to the provided string, or nil if it is an empty string.
```

**Review by anuraaga — APPROVED** [review]
Thanks, looks like there's a small build failure and one doc comment, but otherwise LGTM

**Comment by anuraaga** [comment]
Looks like just a flaky loadtest - @pjanotti @open-telemetry/collector-maintainers do you mind rerunning it and merging if things look ok? Thanks!

**Review by pjanotti — APPROVED** [review]

**Pull request merged**